db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -1613,7 +1627,8 @@ Status DBImpl::FlushMemTable(ColumnFamilyData* cfd,
     }
     WaitForPendingWrites();
 
-    if (!cfd->mem()->IsEmpty() || !cached_recoverable_state_empty_.load()) {
+    if ((!cfd->mem()->IsEmpty() || !cached_recoverable_state_empty_.load()) &&
+        flush_reason != FlushReason::kFlushNoSwitchMemtable) {
```

Comment:
Should we assert here that ```cfd->imm()->NumNotFlushed() > 0``` here if ```flush_reason == FlushReason::kFlushNoSwitchMemtable```?

Reply:
> Should we assert here that `cfd->imm()->NumNotFlushed() > 0` here if `flush_reason == FlushReason::kFlushNoSwitchMemtable`?

If there is no new KV-pair being inserted since last switch. Do we still generate an empty immutable Memtable? If so, it is better to assert `cfd->imm()->NumNotFlushed() > 0` here if `flush_reason == FlushReason::kFlushNoSwitchMemtable`

Reply:
I don't think an empty immutable memtable is generated. The assert I'm suggesting is to just make sure there is something to do, i.e if ```SwitchMemtable``` is not called, then atleast there should be 1 unflushed immutable memtable. That should usually be the case since its a flush error that would have caused the ```kFlushNoSwitchMemtable``` reason.